File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.go
func consecutiveSetBits(n int) bool {
    cnt, p := 0, -1
    for ; n > 0; n >>= 1 {
        r := n % 2
        if r == 1 && r == p {
            cnt++
        }
        p = r
    }
    return cnt == 1
}

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.java
class Solution {
    public boolean consecutiveSetBits(int n) {
        int cnt = 0;
        int p = -1;
        while (n != 0) {
            int r = n % 2;
            if (r == 1 && r == p)
                cnt++;
            n >>= 1;
            p = r;
        }
        return cnt == 1;
    }
}